src/rulesets/oas3/index.json

```diff
@@ -78,6 +78,111 @@
         }
       }
     },
+    "valid-oas-example-in-parameters": {
+      "description": "Examples must be valid against their defined schema.",
+      "message": "{{error}}",
+      "recommended": true,
+      "severity": 0,
+      "type": "validation",
+      "given": "$..parameters..[?(@.example && @.schema)]",
```

Comment:
Paths are too loose.

Reply:
@P0lip, what path for OAS3 `schema` next to `example` in `parameters` you'd rather see? I can't think of anything better atm

Reply:
I don't really know, but this is going to be utterly slow.
@philsturgeon can you think of anything better?

Reply:
I was wondering about this. Would it be better performance wise to:

1. run on all parameters then check for example and schema keys
2. run on all parameters that have an example then check for schema

Or is it the `$..parameters` bit that's the problem? There is an issue where #272 points out there are different types of parameters in the [Link Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#linkObject) so we don't want to double up on that mistake.

Reply:
> Or is it the $..parameters bit that's the problem? 

Yeah, it's `$..parameters` that is the problem, if we could narrow it down a bit, that would be great.
I don't think we are interested in parameters under info object, are we?
If I am not mistaken, parameters are bound to an endpoint/path, so we could at least have paths there, i.e. `$.paths..parameters`.
I truly doubt we aim to validate every property with `parameters` key, so let's limit the scope as much as we can.